Gems/AudioSystem/Code/Source/Editor/AudioControlsEditorWindow.cpp

```diff
@@ -201,15 +201,19 @@ namespace AudioControls
     //-------------------------------------------------------------------------------------------//
     void CAudioControlsEditorWindow::RefreshAudioSystem()
     {
-        QString sLevelName = GetIEditor()->GetLevelName();
-
-        if (QString::compare(sLevelName, "Untitled", Qt::CaseInsensitive) == 0)
+        if (auto audioSystem = AZ::Interface<Audio::IAudioSystem>::Get();
+            audioSystem != nullptr)
         {
-            // Rather pass empty QString to indicate that no level is loaded!
-            sLevelName = QString();
-        }
+            QString sLevelName = GetIEditor()->GetLevelName();
+
+            if (QString::compare(sLevelName, "Untitled", Qt::CaseInsensitive) == 0)
+            {
+                // Rather pass empty QString to indicate that no level is loaded!
+                sLevelName = QString();
+            }
 
-        Audio::AudioSystemRequestBus::Broadcast(&Audio::AudioSystemRequestBus::Events::RefreshAudioSystem, sLevelName.toUtf8().data());
+            audioSystem->RefreshAudioSystem(sLevelName.toUtf8().data());
+        }
```

Comment:
I assuming the prefab developers have some kind of way to remove the dependency on a level from the Audio system.
@AMZN-daimini, @srikappa-amzn or @chiyenteng any thoughts?

Reply:
This is a bit of legacy level functionality in terms of what Audio system is doing here.  The audio system will likely need to update to account for non-levels in the future and work better with prefabs.

Reply:
I've taken a quick look at the RefreshAudioSystem code. It looks like its tightly coupled with the name of the level. It is possible for prefabs to fetch the filepath that corresponds to the currently loaded level. However, if this logic isn't reworked  to function with any prefab, I don't see a point in doing that since we will just be fetching the level name from once place instead of another. If you want to discuss the integration with prefabs, feel free to reach out to me.

Reply:
Spoke with @srikappa-amzn about this.  I think the audio system's dependency on level names/scopes is something that can be removed entirely, but doesn't need to happen here.  There are ways to attach things like bank-loading/unloading to a prefab that fit more closely with how prefabs and components already work.  Pulling that out now will add quite a few more changes to this review, and I think that could be done separately as its own task.